compliance/index/v4/node.go

```diff
@@ -0,0 +1,185 @@
+		Distributions: map[string]*claircore.Distribution{},
+		Repositories:  map[string]*claircore.Repository{},
+		Files:         map[string]claircore.File{},
+	}
+
+	layer, err := constructLayer(ctx, layerDigest, env.NodeScanningV4HostPath.Setting())
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if tmpErr := layer.Close(); tmpErr != nil {
+			err = tmpErr
+		}
+	}()
+
+	reps, err := runRepositoryScanner(ctx, layer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to run repository scanner")
+	}
+
+	pcks, err := runPackageScanner(ctx, layer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to run package scanner")
+	}
+
+	ir, err := coalesceReport(ctx, layerDigest, reps, pcks)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to coalesce report")
+	}
+
+	report = controller.MergeSR(report, []*claircore.IndexReport{ir})
+	report.Success = true
+	report.State = controller.IndexFinished.String()
+
+	r, err = mappers.ToProtoV4IndexReport(report)
+	if err != nil {
+		return nil, errors.Wrap(err, "converting clair report to v4 report")
+	}
+
+	return
+}
+
+func coalesceReport(ctx context.Context, layerDigest string, reps []*claircore.Repository, pcks []*claircore.Package) (*claircore.IndexReport, error) {
+	ch, err := claircore.ParseDigest(layerDigest)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing Clair Core digest")
+	}
+
+	la := &ccindexer.LayerArtifacts{
+		Hash: ch,
+	}
+	la.Repos = append(la.Repos, reps...)
+	la.Pkgs = append(la.Pkgs, pcks...)
+	artifacts := []*ccindexer.LayerArtifacts{la}
+	coal := new(rhel.Coalescer)
+
+	ir, err := coal.Coalesce(ctx, artifacts)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to coalesce report")
+	}
+	return ir, nil
+}
+
+func runPackageScanner(ctx context.Context, layer *claircore.Layer) ([]*claircore.Package, error) {
+	rpm := rpm2.Scanner{}
+	pck, err := rpm.Scan(ctx, layer)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to invoke RPM scanner")
+	}
+	pck = filterPackages(ctx, pck)
+	if pck != nil {
+		log.Infof("Num packages found: %d", len(pck))
+	}
+	for i, p := range pck {
+		p.ID = fmt.Sprintf("%d", i)
+	}
+	return pck, nil
+}
+
+// As we're only interested in the effective running RPM DB,
+// we filter out packages from other DBs like rpm-ostree
+func filterPackages(_ context.Context, pck []*claircore.Package) []*claircore.Package {
+	var filtered []*claircore.Package
+	for _, pkg := range pck {
+		if pkg.PackageDB == "sqlite:usr/share/rpm" {
+			filtered = append(filtered, pkg)
+		}
+	}
+	return filtered
+}
+
+func runRepositoryScanner(ctx context.Context, l *claircore.Layer) ([]*claircore.Repository, error) {
+	c := http.DefaultClient
+	sc := rhel.RepositoryScanner{}
+	config := rhel.RepositoryScannerConfig{
+		DisableAPI:         false,
+		API:                "https://catalog.redhat.com/api/containers/",
+		Repo2CPEMappingURL: "https://access.redhat.com/security/data/metrics/repository-to-cpe.json",
```

Comment:
Realize this has already been merged - but wanted to call out that these URLs should point internally to ACS to support offline / disconnected environments.